src/main/webapp/app/exercises/shared/exercise-scores/exercise-scores-export-button.component.ts

```diff
@@ -110,6 +170,69 @@ export class ExerciseScoresExportButtonComponent {
             }) || []
         );
     }
+
+    /**
+     * Retrieves a list of test cases names contained in a result's feedback list
+     * @param results list of results to extract the test case names from
+     * @private
+     */
+    private getTestCaseNamesFromResults(results: ResultWithPointsPerGradingCriterion[]): string[] {
+        const testCasesNames: Set<string> = new Set();
+        results.forEach((result) => {
+            if (!result.result.feedbacks) {
+                return [];
+            }
+            result.result.feedbacks.map((f) => {
+                testCasesNames.add(f.text ? f.text : 'Test ' + result.result.feedbacks?.indexOf(f) + 1);
+            });
+        });
+        return Array.from(testCasesNames);
+    }
+
+    /**
+     * Extracts test case results from a given result and returns them.
+     * If no feedback is found in the result an empty array is returned
+     * @param result from which the test case results should be extracted
+     * @param testCaseNames list containing the test names
+     * @param withFeedback if true, the feedback's full text is included in case of failed test case
+     * @private
+     */
+    private getTestCaseResults(result: ResultWithPointsPerGradingCriterion, testCaseNames: string[], withFeedback?: boolean): TestCaseResult[] {
+        const testCaseResults: TestCaseResult[] = [];
+
+        testCaseNames.forEach((testCaseName) => {
```

Comment:
```suggestion
        testCaseNames.forEach((testName) => {
```
With this name change, you could leave out the "testName:" in the objects below

Reply:
Added.. Nice, thanks 👍